sources/main.cpp

```diff
@@ -207,6 +207,27 @@ int main(int argc, char* argv[]) {
     ButtonHandler* buttonHandler = new ButtonHandler(hwFactory->getInterruptHandler(), yioapi);
     qmlRegisterSingletonType<ButtonHandler>("ButtonHandler", 1, 0, "ButtonHandler", &ButtonHandler::getQMLInstance);
 
+    // reset combination was pressed on startup
+    if (buttonHandler->resetButtonsPressed()) {
+        // create marker file
+        QFile file("/firstrun");
+        if (file.open(QIODevice::WriteOnly)) {
```

Comment:
If the marker file cannot be written then bail out with an error message, otherwise we'll end up in an inconsistent state!

Reply:
Got it!